src/helpers/functions.ts

```diff
@@ -58,3 +61,54 @@ export const waterfall = <TFns extends ((arg?: any) => any)[]>(
     return chain;
   };
 };
+
+type ParseErr = string;
+export const parseFnData = async (
+  data: unknown,
+  api: InngestAPI
+): Promise<Result<FnData, ParseErr>> => {
+  try {
+    const result = fnDataSchema.parse(data);
+
+    if (result.use_api) {
+      const [evtResp, stepResp] = await Promise.all([
+        api.getRunBatch(result.ctx?.run_id as string),
+        api.getRunSteps(result.ctx?.run_id as string),
```

Comment:
Is it worth checking `result.ctx.run_id` at the start of the function and returning an error if we somehow have an unidentifiable run?